Handle ErrUnexpectedEndOfDemo gracefully.

JanEricNitschke opened the pull request:
Closes https://github.com/pnxenopoulos/awpy/issues/189

We now just print a warning that the error occured and to check the result if we get a `ErrUnexpectedEndOfDemo` when parsing a demo.

Also did a minor cleanup and fixed the gitignore when it comes to demo files.

pnxenopoulos reviewed (approved):
Guessing the changes in `parse_demo.go` seem like fixing redundant logic on my part, yea?

JanEricNitschke commented:
Yeah, the else branch was doing something that was also done at the end of the if branch. So i just moved that out of the if/else and deleted the else branch.
